src/Interpreters/ActionsDAG.cpp

```diff
@@ -1941,4 +1942,102 @@ ActionsDAGPtr ActionsDAG::cloneActionsForFilterPushDown(
     return actions;
 }
 
+static bool chainPreservesSorting(const ActionsDAG::Node* chain)
+{
+    const Field field{};
+    const ActionsDAG::Node* node = chain;
+    while (node)
+    {
+        if (node->type == ActionsDAG::ActionType::FUNCTION)
+        {
+            auto func = node->function_base;
+            if (func)
+            {
+                if (!func->hasInformationAboutMonotonicity())
+                    return false;
+
+                const auto & types = func->getArgumentTypes();
+                if (types.empty())
+                    return false;
+
+                /// TODO: we support monotonicity check only for functions with one parameter but ...
+                ///       if one parameter is variable and other are constant then we can try to check monotonicity as well
+                const auto monotonicity = func->getMonotonicityForRange(*types.front(), field, field);
```

Comment:
in all three cases: here, at l.1974 and l.1991 imo should be provided some comments, why only `front()` is considered.